index.d.ts

```diff
@@ -0,0 +1,39 @@
+declare namespace kdljs {
+  /**
+   * A {@link https://github.com/kdl-org/kdl/blob/main/SPEC.md#document|Document}.
+   */
+  export type Document = Node[];
+
+  /**
+   * A {@link https://github.com/kdl-org/kdl/blob/main/SPEC.md#node|Node}.
+   */
+  export interface Node {
+    /** The name of the Node */
+    name: string;
+    /** Collection of {@link https://github.com/kdl-org/kdl/blob/main/SPEC.md#argument|Arguments} */
+    values: Value[];
+    /** Collection of {@link https://github.com/kdl-org/kdl/blob/main/SPEC.md#property|Properties} */
+    properties: Record<string, Value>;
+    /** Nodes in the {@link https://github.com/kdl-org/kdl/blob/main/SPEC.md#children-block|Children block} */
+    children: Document;
+  }
+
+  /**
+   * A {@link https://github.com/kdl-org/kdl/blob/main/SPEC.md#value|Value}.
+   */
+  export type Value = string | number | boolean | null;
+
+  export interface ParseResult {
+    /** Parsing errors */
+    errors: chevrotain.IRecognitionException[];
```

Comment:
I don't know very much about TypeScript; does this need to be imported?

Reply:
It's a bit weird but no, it doesn't need to be imported.

I just double checked to be sure—`chevotrain` uses [`export as namespace chevrotain`](https://github.com/SAP/chevrotain/blob/01753e36dc2bef849a35a825684b641e0e3508e6/packages/chevrotain/api.d.ts#L1) which exposes the types like a global UMD module ([see TypeScript docs](https://www.typescriptlang.org/docs/handbook/modules.html#umd-modules)).